staging/src/k8s.io/apimachinery/pkg/api/validate/limits_test.go

```diff
@@ -23,53 +23,54 @@ import (
 
 	"k8s.io/apimachinery/pkg/api/operation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	fldTesting "k8s.io/apimachinery/pkg/util/validation/field/testing"
 )
 
 func TestMaxLength(t *testing.T) {
 	cases := []struct {
-		value string
-		max   int
-		err   string // regex
+		name     string
+		value    string
+		max      int
+		wantErrs field.ErrorList // regex
 	}{{
-		value: "",
-		max:   0,
+		name:     "empty string",
+		value:    "",
+		max:      0,
+		wantErrs: nil,
 	}, {
+		name:  "zero length",
 		value: "0",
 		max:   0,
-		err:   "fldpath: Invalid value.*must be no more than",
+		wantErrs: field.ErrorList{
+			field.Invalid(field.NewPath("fldpath"), "0", ".*must be no more than").WithOrigin("maxLength"),
```

Comment:
I'm not convinced we need a non-nil value here, but you are closer to this test, so your call :)